main.go

```diff
@@ -334,6 +335,7 @@ func startHTTPServer(spec *types.SubmarinerSpecification) *http.Server {
 	srv := &http.Server{Addr: ":" + spec.MetricsPort, ReadHeaderTimeout: 60 * time.Second}
 
 	http.Handle("/metrics", promhttp.Handler())
+	http.HandleFunc("/debug", pprof.Profile)
```

Comment:
Does this enable profiling unconditionally even in production (ie.., non debug builds)? Generally profiling is enabled to investigate any performance issues and is turned off once we are done. 

Any idea about the overhead it introduces in the system?

Reply:
> I’ve updated the commit message:
> 
> > The performance impact is minimal, except when streaming CPU profiles; that only happens on demand, e.g. with a query on .../debug/pprof/profile?seconds=30 (for a 30-second profile). See https://medium.com/google-cloud/continuous-profiling-of-go-programs-96d4416af77b which explains how Google run with permanent profiling; the 5% cost (amortised to 0.5%) is incurred by streaming profiles periodically, and doesn't apply to this change.

The idea here is to provide permanent access to profiling features. This will allow us (with appropriate permissions) to examine heap problems, blocked threads etc. in production setups without having to reproduce the trigger.

Reply:
Cool, thanks.